src/include/schema.h

```diff
@@ -256,6 +259,39 @@ struct __wt_table {
 	}								\
 } while (0)
 
+/*
+ * WT_WITH_HOTBACKUP_READ_LOCK, WT_WITH_HOTBACKUP_WRITE_LOCK --
+ * 	Acquire the hot backup lock, perform an operation, drop the lock.
+ */
+#define	WT_WITH_HOTBACKUP_READ_LOCK(session, op) do {			\
+	WT_CONNECTION_IMPL *__conn = S2C(session);			\
+	if (F_ISSET(session, WT_SESSION_LOCKED_HOTBACKUP)) {		\
+		op;							\
+	} else {							\
+		__wt_readlock(session, &__conn->hot_backup_lock);	\
+		F_SET(session, WT_SESSION_LOCKED_HOTBACKUP_READ);	\
```

Comment:
If you made the following line `if (!conn->hot_backup)` then almost every multi-line issue and lots of wrapper functions go away.

Reply:
This approach won't work for `__checkpoint_lock_dirty_tree_int` from `txn_ckpt.c`.
I also feel that it'd be misleading to have a macro that doesn't necessarily execute its given statement.

I could have a macro just for that check and call it like:
```
WT_WITH_HOTBACKUP_READ_LOCK(session,
    WT_IFN_HOTBACKUP(ret = __wt_ftruncate(session, block->fh, len)));
```
But it seems to be not much of a win over just having the condition there.